auth/user_mgt.go

```diff
@@ -0,0 +1,655 @@
+		delete(p.payload, payloadName)
+	}
+	return nil
+}
+
+func processClaims(p *commonParams, payloadName, errorName string) error {
+	if _, ok := p.payload[payloadName]; !ok {
+		return nil
+	}
+	if err := p.checkReservedClaims(); err != nil {
+		return err
+	}
+	return p.setCustomAttributes()
+}
+
+func (p *commonParams) checkReservedClaims() error {
+	cc := p.payload["customClaims"]
+	switch claims := cc.(type) {
+	case map[string]interface{}:
+		for _, key := range reservedClaims {
+			if _, ok := claims[key]; ok {
+				return fmt.Errorf("CustomClaims, claim %q is reserved, and must not be set", key)
+			}
+		}
+	default:
+		return fmt.Errorf("CustomClaims: unexpected type")
+	}
+	return nil
+}
+
+func (p *commonParams) setCustomAttributes() error {
+	cc := p.payload["customClaims"]
+	b, err := json.Marshal(cc)
+	if err != nil {
+		return fmt.Errorf("CustomClaims Marshaling error: %v", err)
+	}
+	s := string(b)
+	if s == "null" {
+		s = "{}"
+	}
+	p.payload["customAttributes"] = s
+	delete(p.payload, "customClaims")
+	return nil
+}
+
+func (p *commonParams) addToListParam(listname, param string) {
+	if _, ok := p.payload[listname]; ok {
+		p.payload[listname] = append(p.payload[listname].([]string), param)
+	} else {
+		p.set(listname, []string{param})
+	}
+}
+
+// Validators.
+
+// No validation needed. used for bool fields
+func allowed(p *commonParams, _, _ string) error {
+	return nil
+}
+
+func nonEmpty(p *commonParams, fieldName, errorName string) error {
+	if val, ok := p.payload[fieldName]; ok {
+		if len(val.(string)) == 0 {
+			return fmt.Errorf("%s must be a non-empty string", errorName)
+		}
+	}
+	return nil
+}
+
+func validEmail(p *commonParams, fieldName, errorName string) error {
+	if val, ok := p.payload[fieldName]; ok {
+		if parts := strings.Split(val.(string), "@"); len(parts) != 2 || len(parts[0]) == 0 || len(parts[1]) == 0 {
+			return fmt.Errorf("malformed %s string: %q", errorName, val)
+		}
+	}
+	return nil
+}
+
+func strlenGTE(i int) func(*commonParams, string, string) error {
+	return func(p *commonParams, fieldName, errorName string) error {
+		if val, ok := p.payload[fieldName]; ok {
+			if len(val.(string)) < i {
+				return fmt.Errorf("invalid %s string. %s must be a string at least %d characters long", errorName, errorName, i)
+			}
+		}
+		return nil
+	}
+}
+
+func strlenLTE(i int) func(*commonParams, string, string) error {
+	return func(p *commonParams, fieldName, errorName string) error {
+		if val, ok := p.payload[fieldName]; ok {
+			if len(val.(string)) > i {
+				return fmt.Errorf("%s must be a string at most %d characters long", errorName, i)
+			}
+		}
+		return nil
+	}
+}
+
+func validPhone(p *commonParams, fieldName, errorName string) error {
+	if val, ok := p.payload[fieldName]; ok {
+		if !regexp.MustCompile(`\+.*[0-9A-Za-z]`).MatchString(val.(string)) {
+			return fmt.Errorf(
+				"invalid %s: %q. %s must be a valid, E.164 compliant identifier",
```

Comment:
Please get rid of the errorName argument. It is clear it's phone number in this case. Same goes for most other validators in this file.